### ISSUE TITLE
Queries are hitting 32 filter limit prematurely.

### ISSUE THREAD

**Issue by SaraMoad** [elastic/search-ui#837]
**Describe the bug**
- I am trying to use an array to filter which records should be returned.
- I am passing an array of ~500 values in. I am getting the error 'An unexpected error occurred: [400] Filters cannot have more than 32 filters'
- I am have done so trying to follow the guidance laid out [here](https://www.elastic.co/guide/en/app-search/current/filters.html#filters-arrays-vs-objects.): 

**To Reproduce**
` const config: SearchDriverOptions = {
    searchQuery: {
      facets: buildFacetConfigFromConfig(),
      disjunctiveFacets: ["level", "castingtime", "rarity", "ispfslegal", "traits", "subtype", "school", "area", "duration", "range", "savingthrow", "targets"],
      ...buildSearchOptionsFromConfig(),
      filters: [
        {
          field: "nethysid",
          type: "any",
          values: spellsToShowIDs
        }
      ]
    },
    autocompleteQuery: buildAutocompleteQueryConfig(),
    apiConnector: connector,
    alwaysSearchOnInitialLoad: true
  };`

**Expected behavior**
I expect the array spellsToShowIDs to filter successfully.

**Screenshots**
We can see the array size that i'm using, and the log of the config object here:
![image](https://user-images.githubusercontent.com/102942360/182867089-8326f23d-be56-42d1-9faa-bf8034dc2cc5.png)
We can see the request here:
![image](https://user-images.githubusercontent.com/102942360/182867032-f658df36-e683-485a-b9ca-f43151fe8e3c.png)

**Which backends and packages are you using:**
Backend: [App Search]
Packages: [react-search-ui, search-ui-app-search-connector]

**Comment by joemcelroy** [comment]
thanks for the report! Will try get back to you, i suspect this is an issue with how we unroll filters together.  Linked with https://github.com/elastic/search-ui/issues/763

**Comment by JamesLove002** [comment]
I’m also stuck with this issue!!

**Comment by wentaoxu415** [comment]
HI @joemcelroy, is there any update on the fix for this issue?

**Comment by joemcelroy** [comment]
no update yet @wentaoxu415. I will try get this prioritized.

**Comment by joemcelroy** [comment]
I've started looking into this bug. Few things want to highlight:

- We are currently creating a new "filter" per value, even if it was grouped together inside values. This hits a ceiling fast with only 32 filters can be applied. 
- we should be using one filter per field with an **array** of values. https://www.elastic.co/guide/en/app-search/current/filters.html#filters-arrays-all-none
- ranges will not be able to be grouped together. Requires a filter per range type

I've opened a PR which should solve the problem. Hopefully next week we can publish a release.

**Comment by JasonStoltz** [comment]
I think the reasoning is that these two are not necessarily functionally equivalent.


initial must match "additional values" AND "and values" AND "and even more values"
```
        all: [
          { initial: "additional values" },
          { initial: "and values" },
          { initial: "and even more values" }
        ]
```

initial must match "additional values" OR "and values" OR "and even more values"
```
        all: [
          {
            initial: ["additional values", "and values", "and even more values"]
          }
        ]
```

I think that as soon as you use the array syntax, the values implicitly become "OR"d. Let me do some quick testing to verify this.

**Comment by JasonStoltz** [comment]
So, I can confirm with the following requests:

The following matches a total of 12 national parks using our test data set, any park in Alaska OR California
```
{
  "query": "",
  "filters": {
      "all": [{
          "states": ["Alaska", "California"]
      }]
  }
}
```

The following matches a total of 0 national parks using our test data set, because there is no park in Alaska AND California
```
{
  "query": "",
  "filters": {
      "all": [{
          "states": ["Alaska"]
      },
      {
          "states": ["California"]
      }]
  }
}
```

**Comment by JasonStoltz** [comment]
```
{
  "query": "",
  "filters": {
      "all": [{
          "states": ["Alaska", "California"]
      }]
  }
}
```

Is functionally equivalent to:
{
  "query": "",
  "filters": {
      "all": [{
          "any": [{
            "states": ["Alaska"]
          },{
            "states": ["California"]
          }]
      }]
  }
}

**Comment by JasonStoltz** [comment]
Whoops, those comments were meant for the PR.

**Comment by JasonStoltz** [comment]
> we should be using one filter per field with an array of values. https://www.elastic.co/guide/en/app-search/current/filters.html#filters-arrays-all-none

We can use the array syntax as long as folks are applying filters as "any" (OR). We can't use the same array method to group values if they are meant to be "all" (AND).

The 32 filter limit is [hardcoded in App Search](https://www.elastic.co/guide/en/app-search/8.4/limits.html). Many of these limits [can be overridden](https://www.elastic.co/guide/en/app-search/8.4/limits.html), but unfortunately not the filter limit.

I think for the most part though, most folks applying a large number of filters like we see above will be applying them as an "any" filter, not "all", so our PR should help.

**Comment by joemcelroy** [comment]
now been released under v1.18.1 🎉 This change will rollup the filters into a single multi value OR or NOT type filter. an AND type filter still will need a filter per value.
